internal/manifests/collector/ports.go

```diff
@@ -213,12 +222,16 @@ func getTracesReceiversServicePorts(logger logr.Logger, config *adapters.CwaConf
 	return tracesPorts
 }
 
-func getAppSignalsServicePortsMap() map[int32][]corev1.ServicePort {
-	servicePortMap := make(map[int32][]corev1.ServicePort)
+func addAppSignalServicePorts(servicePortsMap map[int32][]corev1.ServicePort) {
 	for k, v := range AppSignalsPortToServicePortMap {
-		servicePortMap[k] = v
+		servicePortsMap[k] = v
+	}
+}
+
+func getApplicationSignalsReceiversServicePorts(config *adapters.CwaConfig, servicePortsMap map[int32][]corev1.ServicePort) {
+	if isAppSignalEnabled(config) {
+		addAppSignalServicePorts(servicePortsMap)
```

Comment:
why not just use `getReceiverServicePort` to follow the same pattern as before, and then we can have the same error checks if there are duplicate ports?
```
getReceiverServicePort(logger, ":4315", AppSignalsGrpc, corev1.ProtocolUDP, servicePortsMap)
getReceiverServicePort(logger, ":4316", AppSignalsHttp, corev1.ProtocolUDP, servicePortsMap)
getReceiverServicePort(logger, ":2000", AppSignalsProxy, corev1.ProtocolUDP, servicePortsMap)
```

I put the hardcoded endpoints ":4315", ":4316", and ":2000" for the endpoint parsing, but I imagine something like this should work. Maybe have these as the new constants and remove the original map.

Reply:
done!